include/libcyphal/transport/can/delegate.hpp

```diff
@@ -101,6 +101,14 @@ class CanardMemory final : public ScatteredBuffer::IStorage
         return bytes_to_copy;
     }
 
+    void observeFragments(ScatteredBuffer::IFragmentsObserver& observer) const override
```

Comment:
Would be cool if we had something like `IInvokable<...>` defined in CETL and implemented by `cetl::function`, to enhance composability. The user could then either use a custom implementation for this interface, or use a cetl function.

Reply:
Internally at CETL we already have something similar, see
```
namespace detail
{
template <typename Result, typename... Args>
class function_handler : public rtti_helper<function_handler_typeid_t>
{
public:
    virtual Result operator()(Args...) = 0;

};  // function_handler
...
```
but it has its issues which sooner or later we might need try to resolve. I'm talking about `Args... args` arguments which currently could be only copied but not moved (but I can't make it forward-able `Args&&... args`) - not sure yet how to address it, so it's a limitation currently).